src/custom/pages/Claim/index.tsx

```diff
@@ -59,10 +59,22 @@ import { useClaimDispatchers, useClaimState } from 'state/claim/hooks'
 import { ClaimStatus } from 'state/claim/actions'
 import { useAllClaimingTransactionIndices } from 'state/enhancedTransactions/hooks'
 
+import { ApprovalState, useApproveCallbackFromClaim } from 'hooks/useApproveCallback'
+import { OperationType } from 'components/TransactionConfirmationModal'
+import useTransactionConfirmationModal from 'hooks/useTransactionConfirmationModal'
+
+import { GNO } from 'constants/tokens'
+import { CurrencyAmount, MaxUint256 } from '@uniswap/sdk-core'
+import { SupportedChainId } from 'constants/chains'
+import { CheckCircle } from 'react-feather'
+import Row from 'components/Row'
+
+// Max approve amount
+const MAX_GNO_UINT256 = CurrencyAmount.fromRawAmount(GNO[SupportedChainId.RINKEBY], MaxUint256)
```

Comment:
Does it make sense to hard code for each token that'll be approved?
Also, you are relying on rinkeby's GNO. Won't that be a problem?
I know, I know, not likely a problem.

I just thought this would work like approval for any other token before trading

Reply:
good point, it will be an issue on another network..